### PR TITLE
Restricted edit section access to owned sections

### PR DESCRIPTION
**Pull request by kobryan0619**
Currently, users can access sections that they do not own by using the correct URL with section ID that does not belong to them.  This addresses this problem.

## Testing

- This includes one test to make sure users cannot access a section that doesn't belong to them.

### CODE REVIEW

File: dashboard/test/controllers/sections_controller_test.rb [hunk 1]
```diff
@@ -216,4 +219,11 @@ class SectionsControllerTest < ActionController::TestCase
     get :new, params: {loginType: 'word', participantType: 'student'}
     assert_response :success
   end
+
```
Comment (hannahbergam) [inline dashboard/test/controllers/sections_controller_test.rb:219]: Yay! Thank you for adding this!

**Review by hannahbergam — COMMENTED** [review]
This seems good to me! I take it you tried to access the url on localhost just to verify?

File: dashboard/test/controllers/sections_controller_test.rb [hunk 2]
```diff
@@ -16,6 +16,9 @@ class SectionsControllerTest < ActionController::TestCase
 
     @flappy_section = create(:section, user: @teacher, login_type: 'word', script_id: Unit.flappy_unit.id)
     @flappy_user_1 = create(:follower, section: @flappy_section).student_user
+
+    @sections = create_list :section, 3, user: @teacher
+    @section = @sections.first
```
Comment (bethanyaconnor) [inline dashboard/test/controllers/sections_controller_test.rb]: I don't think these are needed?

**Review by bethanyaconnor — APPROVED** [review]

File: dashboard/test/controllers/sections_controller_test.rb (hunk 2 above)
Comment (kobryan0619) [reply dashboard/test/controllers/sections_controller_test.rb]: Yes, that is correct.  I removed them and re-tested just to check.

**Pull request merged**